### PR TITLE
HBASE-25174 Remove deprecated fields in HConstants

### PR DESCRIPTION
**Pull request by HorizonNet**
Remove the deprecated fields, which can be removed in 3.0.0. Marked the
constant OLDEST_TIMESTAMP as InterfaceAudience.Private as it is only use
in classes, which are also marked as InterfaceAudience.Private.

### CODE REVIEW

**Review by virajjasani — APPROVED** [review]
+1, nice cleanup of much needed class (pending QA)

File: hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java [hunk 1]
```diff
@@ -661,10 +660,8 @@
    * Timestamp to use when we want to refer to the oldest cell.
    * Special! Used in fake Cells only. Should never be the timestamp on an actual Cell returned to
    * a client.
-   * @deprecated Should not be public since hbase-1.3.0. For internal use only. Move internal to
-   *   Scanners flagged as special timestamp value never to be returned as timestamp on a Cell.
    */
-  @Deprecated
+  @InterfaceAudience.Private
   public static final long OLDEST_TIMESTAMP = Long.MIN_VALUE;
```
Comment (Apache9) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java:665]: Let's move it to another place?

**Review by Apache9 — CHANGES_REQUESTED** [review]

File: hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java (hunk 1 above)
Comment (HorizonNet) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java:665]: Makes sense. I looked through the usages and there doesn't seem to be a common place. Do we have a place for kind of "private" constants?

Comment (HorizonNet) [reply hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java]: Didn't found a good place for this one and created a new class for constants intended for private use.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 11s |  master passed  |
| +1 :green_heart: |  compile  |   4m 10s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 40s |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m 56s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 12s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 58s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m  4s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m 13s |  hbase-server: The patch generated 3 new + 55 unchanged - 2 fixed = 58 total (was 57)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  20m 24s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  spotbugs  |   3m 42s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 22s |  The patch does not generate ASF License warnings.  |
|  |   |  57m 35s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux c8b74d5a01dc 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| Max. process+thread count | 86 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 52s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m 44s |  master passed  |
| +1 :green_heart: |  compile  |   1m 56s |  master passed  |
| +1 :green_heart: |  shadedjars  |  10m 52s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m 22s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 18s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  7s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 16s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |  10m  5s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  8s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m  4s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 146m 12s |  hbase-server in the patch passed.  |
|  |   | 192m  7s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux c06bf0ff30d1 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/testReport/ |
| Max. process+thread count | 4224 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 21s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 21s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 13s |  master passed  |
| +1 :green_heart: |  compile  |   1m 26s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 29s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 57s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 23s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 49s |  hbase-common in the patch passed.  |
| -1 :x: |  unit  | 215m 43s |  hbase-server in the patch failed.  |
|  |   | 251m 36s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0245db264621 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/artifact/yetus-jdk8-hadoop3-check/output/patch-unit-hbase-server.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/testReport/ |
| Max. process+thread count | 3593 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 32s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 24s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  0s |  master passed  |
| +1 :green_heart: |  compile  |   4m  9s |  master passed  |
| +1 :green_heart: |  checkstyle  |   1m 30s |  master passed  |
| +1 :green_heart: |  spotbugs  |   2m 51s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 38s |  the patch passed  |
| +1 :green_heart: |  compile  |   4m  7s |  the patch passed  |
| +1 :green_heart: |  javac  |   4m  7s |  the patch passed  |
| -0 :warning: |  checkstyle  |   1m  3s |  hbase-server: The patch generated 3 new + 55 unchanged - 2 fixed = 58 total (was 57)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  22m  7s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1 3.3.0.  |
| +1 :green_heart: |  spotbugs  |   4m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 27s |  The patch does not generate ASF License warnings.  |
|  |   |  59m 38s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | dupname asflicense javac spotbugs hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 3fa57286e412 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| checkstyle | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/artifact/yetus-general-check/output/diff-checkstyle-hbase-server.txt |
| Max. process+thread count | 96 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/console |
| versions | git=2.17.1 maven=3.6.3 spotbugs=4.2.2 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 14s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 55s |  master passed  |
| +1 :green_heart: |  compile  |   1m 43s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 28s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 44s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 44s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 27s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   1m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 21s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 206m 55s |  hbase-server in the patch passed.  |
|  |   | 245m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 0e11d4950b9e 4.15.0-136-generic #140-Ubuntu SMP Thu Jan 28 05:20:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/testReport/ |
| Max. process+thread count | 2737 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 28s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 12s |  master passed  |
| +1 :green_heart: |  compile  |   1m 27s |  master passed  |
| +1 :green_heart: |  shadedjars  |   8m 25s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 58s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 14s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  javac  |   1m 28s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 21s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 56s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   1m 50s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  unit  | 216m  9s |  hbase-server in the patch passed.  |
|  |   | 251m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/artifact/yetus-jdk8-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/2558 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 27144fef3189 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 7a31557c51 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/testReport/ |
| Max. process+thread count | 3752 (vs. ulimit of 30000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-2558/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Pull request merged**